### ISSUE TITLE
[All source games] Steam auth exploit/harvest by scammers

### ISSUE THREAD

**Issue by Zeitgesit** [ValveSoftware/Source-1-Games#3857]
Since now 1 year, there is an en-masse scam operation taking place on all Source based games platforms.
The server is called Fastpath

They have found a way to exploit players steam auth tickets, keeping them long after the players have disconnected.
Making their servers look packed. 
On each game platform they have created about 15 fake silent redirect servers. Rendering this a mass scale operation.
This on repeat is emptying all legitimate servers. Making the few real ones impossible to find, lost in a sea of fake servers.

This operation has many facets, 
I will list a few here:
- All players names are copied and re assigned to bots once they leave
- All their fake server names are names copied from other servers
- Monies requested for being unbanned
- Threats of shutting down servers if monies are not received

Could the exploit of steam auth tickets be patched.
Could the scammer's IP addresses be banned from Source games
![stolenservernames](https://user-images.githubusercontent.com/66415427/155857017-67777bea-f05a-4bc4-b359-6faa0be16955.jpg)
![website](https://user-images.githubusercontent.com/66415427/155857020-b459c60c-bb10-4b49-8e11-822a97ec1a51.jpg)

**Comment by Zeitgesit** [comment]
Today on the HL2DM platform, a total of 300 sotolen steam auth tickets distributed over 30 fake servers and many players trapped.
Legitimate servers are now completely empty for most.
It is now official that fastpath has completely destroyed years of work for server owners on HL2DM and the platform itself.
I am saddened that this topic has not been picked up.

**Comment by Combine900** [comment]
@kisak-valve Can this thread get attension?

**Comment by worMatty** [comment]
@Combine900 can you please stop bumping issues. It won't bring fixes any sooner. All you are doing is annoying those of us who are subscribed to this repository.

**Comment by Zeitgesit** [comment]
@kisak-valve I am sorry to do this, but I have been trying to get this resolved since now over two years.
Apologies if this annoyes you, but our game has been destroyed since two years and not a finger was lifted to help fixing it.
The servers with 7 players are all **fake redirect** servers.
The server on top is the scammer server with all these fake servers farming all possible new players into the abyss.
This is now a desperate call for help.

**Thank you**

![scamcontinues](https://user-images.githubusercontent.com/66415427/191478337-49864cc2-2729-4ba8-84c4-3a3d2925498c.jpg)

**Comment by kisak-valve** [comment]
Hello @Zeitgesit, friendly reminder that I'm a moderator for Valve's issue trackers on Github, and not a Valve developer myself. We'll need to hear from a relevant dev for any insight into this issue.

**Comment by Zeitgesit** [comment]
@kisak-valve Thank you for taking the time to reply to me I really appreciate it.
Do you think we even stand a chance of anyone at Valve considering to pick up this issue? Do you think someone has seen it?

**Comment by ioa1994** [comment]
Sorry to continue bumping this per the comments above, but I reached this post from Google after booting up HL2DM having not played in many years. I accidentally joined a Fastpath server by attempting to join what I thought was a legitimate server, but got redirected. I played there for a few days and slowly realized the level of manipulation taking place at the player/bot level. 


I noticed some players who I knew to be legitimate players, were somedays acting like bots. Sure enough, the "bots" as Zeitgeist explains, are really the names and auth tokens of users who had played on the server previously. Not only that, but I saw one of them post a message to the chat that I had posted the day prior. They are taking user's messages and "replaying" them in the chat as bots in order to appear more authentic. So you have bots saying "ggs" and "has anyone gone trick or treating" and various other messages - it took me until I saw my own message "replayed" that this is what was happening. I am sure my own username is appearing as a user well after I log off, too.


Valve's failure to respond to this issue through community channels and official (github) channels is a tacit confession that they are ready to kill the game, because that's exactly what's going to happen if legitimate server hosts' traffic is redirected to these scummy servers. Just thought I should put in my 2 cents.

**Comment by Zeitgesit** [comment]
I could not agree more with everything you say ioa. It gets worse and worse all the time.
Lately the servers have over 24 fake players in, Each!
Since more than 2 years, the player amount of HL2DM has been reduces drasticlly! There are now more players on HL1DM which is unthinkable.
In anycase, id you look for a good place to play, you know where to find us.

**Comment by Zeitgesit** [comment]
Crickets

**Comment by Zeitgesit** [comment]
Look at the amount of hacking reports on that IP address, and still nothing is done by Steam or anyone else.
https://www.abuseipdb.com/check/164.132.202.2

**Comment by Zeitgesit** [comment]
Good day everyone, and a wonderful start in 2023
I was just wondering if anyone at Valve were going to lift a finger about this devastating problem we have encountered for now **3 years** ?

**Comment by Zeitgesit** [comment]
158.69.22.27 - The scam continues, years down the line, will anything be done?

**Comment by Zeitgesit** [comment]
Hey there Ashetf2, thanks for your comment. Would you be willing to help out with this? Could you otherwise point me to it?
Many thanks!

**Comment by Ashetf2** [comment]
Just create a submission in the Workshop. There are tutorials in the internet. However, you'll need to give your tax info for your submission to be public.

**Comment by Zeitgesit** [comment]
@kisak-valve Sorry to bombard you today! But this issue will soon be 2 years old.
Would you have the capacity to poke a developer about this?
Thanks again!

**Comment by ioa1994** [comment]
> @kisak-valve Sorry to bombard you today! But this issue will soon be 2 years old. Would you have the capacity to poke a developer about this? Thanks again!

Pinging @kisak-valve as well - The game is an absolute graveyard at this point. If Half Life Deathmatch can get the 25 year anniversary update, Half Life 2 Deathmatch should get it too. The details of this issue render the game basically unusable. Would really appreciate your escalating this to the dev team.

**Comment by Zeitgesit** [comment]
Just adding the updated blacklist here useful if you want to find games in your browser.
[server_blacklistDEC2023.zip](https://github.com/ValveSoftware/Source-1-Games/files/13734140/server_blacklistDEC2023.zip)

**Comment by Zeitgesit** [comment]
Adding one of many community discussions.
https://steamcommunity.com/app/320/discussions/0/3114781060460063768/
Apparently they are scamming since 10 years! - we've been affected for 4 years. I woner how long it can take for this to be resolved.

**Comment by dabcodmap** [comment]
it looks like counter strike source has the same problem (i.e., infected by scummy fastpath servers). Any time i want to join another server i get redirected to their servers. is there any way to solve this issue? i'm a bit worried about safety of my own account since they duplicated it to create a bot

**Comment by dabcodmap** [comment]
yes I noticed the chat thing too! imo, if valve and steam don't take action is because they want the source games to slowly die...
quite sad.
Still hope to light up that candle too though

**Comment by Zeitgesit** [comment]
@dabcodmap I am of the same opinion. Yes, let's hope they do something.

**Comment by f-o** [comment]
These guys have now spread to TF2 as well. Exactly same behaviour as described above.

I have had to resort to using community block-lists to fix this issue. Maybe this will be useful for other players as well.

https://steamcommunity.com/sharedfiles/filedetails/?id=2502626384

**Comment by Zeitgesit** [comment]
@f-o This is the blacklist I made which I always update.
https://gamebanana.com/mods/289337

**Comment by Zeitgesit** [comment]
Over 2 years later and still nothing

**Comment by wow32134** [comment]
This is still an occurring problem in the new HL2DM update, Valve pls fix.

**Comment by antielectronbomb** [comment]
This still happens to all Source games in 2025. Something needs to be done. It would be helpful to know what methods these kinds of servers use so Valve developers and community volunteers are able to fix this.

**Comment by Zeitgesit** [comment]
@kisak-valve Do you think we still have a chance for Valve to even consider looking at it? It has been well over 3 years and they seem to not care - perhaps they have no idea about it? Could you help us?

**Comment by dodbits** [comment]
> [@kisak-valve](https://github.com/kisak-valve) Do you think we still have a chance for Valve to even consider looking at it? It has been well over 3 years and they seem to not care - perhaps they have no idea about it? Could you help us?

As a whole issue I think it's been longer than that, at least since 2009.
I wrote up [an article on what I found](https://www.dodbits.com/dods/index.php/source-graphics/game-server-set-up/steam-game-servers-that-use-fake-clients-and-other-bad-behaviour) on it if you are interested. 

One of the items was the interesting differences between some Game Server lists that can see the fake clients and number them.
Then you go to the Steam Game server browser (in game or from the Steam client menu) and you see the same server is NOT reporting it bots.

I pondered if this is actually OK with Steam's online conduct rules. I don't think it is.

As far as avoiding these guys I guess you could say the best way is to use the online lists to find servers then bookmark them in the in game "Favorites" tab.

The ones I know report bots properly (where the Steam browser does not) are...
Game Tracker (it rates your server as "NR" if fake client are detected)
https://www.gametracker.com/games/

Steam Server Browser It does show bots that the Steam Browser does not 
https://www.steamserverbrowser.com/games/half-life-2-deathmatch/servers/164.132.201.109:27017

And for TF2 only this one (again) shows bots where the in game browser does not pick up the exploit used in bot reports...
https://teamwork.tf/community/quickplay/gps/48.8582,2.3387

Using that method does show that certain game server operators are getting around the bot count and other exploits and it does fool a lot of players and also has some side effects that are really... against Valves code of conduct.

**Comment by Zeitgesit** [comment]
Funny enough the scam has stopped, at least on hl2dm - I have not check the other platforms yet. It has stopped for about 10 days now. Not sure if something was done or if they are just preparing an extra 1000 fake servers.

**Comment by dodbits** [comment]
> Funny enough the scam has stopped, at least on hl2dm - I have not check the other platforms yet. It has stopped for about 10 days now. Not sure if something was done or if they are just preparing an extra 1000 fake servers.

Yes, some recent change has occurred. (EDIT: I can't see any proof of that in updates, but it does look changed)

You can now spot a server that is falsifying the "Bots" column to hide information to the players.
The "Players" and "Bots" columns in the browser "Internet" tab is now not matching up with the "Game Info" numbers.

It means only one thing... a breach of the "Steam Online Conduct" rules.

[Image - example of hiding bots...](https://www.dodbits.com/dods/images/stories/game_servers/HL2_DM_hide_bot_icon.jpg)

Some of those "Players" in the Game Info box (edit: _could be_ if you look that name up and match a Steam ID) bots with Steam ID account names ... taken from real Steam ID's.

They can no longer hide. That server and others in other steam games is breaking more than one rule.
